geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradePartitionRegionClearMixedServerPartitionedRegion.java

```diff
@@ -0,0 +1,410 @@
+    } else {
+      System.out.println("running against these versions: " + result);
+    }
+    return result;
+  }
+
+  @Test
+  public void testPutAndGetMixedServerPartitionedRegion() throws Exception {
+    doTestPutAndGetMixedServers(oldVersion);
+  }
+
+  /**
+   * This test starts up multiple servers from the current code base and multiple servers from the
+   * old version and executes puts and gets on a new server and old server and verifies that the
+   * results are present. Note that the puts have overlapping region keys just to test new puts and
+   * replaces
+   */
+  void doTestPutAndGetMixedServers(String oldVersion)
+      throws Exception {
+    VM currentServer1 = VM.getVM(VersionManager.CURRENT_VERSION, 0);
+    VM oldServerAndLocator = VM.getVM(oldVersion, 1);
+    VM currentServer2 = VM.getVM(VersionManager.CURRENT_VERSION, 2);
+    VM oldServer2 = VM.getVM(oldVersion, 3);
+
+    String regionName = "aRegion";
+
+    final String serverHostName = NetworkUtils.getServerHostName();
+    final int port = AvailablePortHelper.getRandomAvailableTCPPort();
+    oldServerAndLocator.invoke(() -> DistributedTestUtils.deleteLocatorStateFile(port));
+    try {
+      final Properties props = getSystemProperties();
+      props.remove(DistributionConfig.LOCATORS_NAME);
+
+      // Fire up the locator and server
+      oldServerAndLocator.invoke(() -> {
+        props.put(DistributionConfig.START_LOCATOR_NAME,
+            "" + serverHostName + "[" + port + "]");
+        props.put(DistributionConfig.ENABLE_CLUSTER_CONFIGURATION_NAME, "false");
+        cache = createCache(props);
+        Thread.sleep(5000); // bug in 1.0 - cluster config service not immediately available
+      });
+
+      props.put(DistributionConfig.LOCATORS_NAME, serverHostName + "[" + port + "]");
+
+      // create the cache in all the server VMs.
+      for (VM vm : Arrays.asList(oldServer2, currentServer1, currentServer2)) {
+        vm.invoke(() -> {
+          cache = createCache(props);
+        });
+      }
+      // spin up current version servers
+      for (VM vm : Arrays.asList(currentServer1, currentServer2)) {
+        vm.invoke(
+            () -> assertVersion(cache, VersionManager.getInstance().getCurrentVersionOrdinal()));
+      }
+
+      // create region
+      for (VM vm : Arrays.asList(currentServer1, currentServer2, oldServerAndLocator, oldServer2)) {
+        vm.invoke(() -> createRegion(cache, regionName));
+      }
+
+      // put some data in the region to make sure there is something to clear.
+      putDataSerializableAndVerify(currentServer1, regionName, currentServer2, oldServerAndLocator,
+          oldServer2);
+
+      // invoke Partition Region Clear and verify we didn't touch the old servers.
+
+      currentServer1.invoke(() -> {
+        assertRegionExists(cache, regionName);
+        PartitionedRegion region = (PartitionedRegion) cache.getRegion(regionName);
+
+        Throwable thrown = catchThrowable(region::clear);
```

Comment:
would be nice to have the exception tell which servers are outdated as a verification.

Reply:
I have changed the exception to ServerVersionMismatchException(members, feature name) it now produces a message "A server's [oldMember] version was too old for : Partitioned Region Clear"

Reply:
Done